src/screenshots/capturer.js

```diff
@@ -60,14 +56,18 @@ export default class Capturer {
             screenshotPath      = joinPath(this.baseScreenshotsPath, Capturer._correctFilePath(customPath));
         }
         else {
-            var path = joinPath(this.baseScreenshotsPath, this.baseDirName, this.testDirName);
+            var testDirName       = `test-${this.testIndex}`;
+            var path              = joinPath(this.baseScreenshotsPath, this.baseDirName, testDirName);
+            var pathForScreenshot = this.quarantineAttemptNum !== null ?
+                                    joinPath(path, `run-${this.quarantineAttemptNum}`) :
+                                    path;
```

Comment:
I don't like `pathForScreenshot` name, BTW. `path` is path for screenshot too. Just get rid of ternary here and make conditional assignment of `joinPath(path, `run-${this.quarantineAttemptNum}`)`